quackfactory/cogs/on_join.py

```diff
@@ -0,0 +1,25 @@
+from discord.ext import commands
+import discord
+import logging
+
+log = logging.getLogger(__name__)
+
+JOIN_MESSAGE_CHANNEL_ID = 784643833488998411
+
+class OnJoin(commands.Cog):
+  def __init__(self, bot):
+    self.bot = bot
+  
+  @commands.Cog.listener()
+  async def on_guild_join(self, guild):

```

Comment:
I like the solution, but as @amazingproducer mentioned, if you are down for it, you can turn this PR into a solution for #5.

This would mean implementing **on_raw_reaction_add** and **on_raw_reaction_remove** events handlers.

Reply:
@hum Work is my main priority as of now, but I'll try to find time and collaborate on developing a solution for this. 

Optimally we should fetch the value `JOIN_MESSAGE_CHANNEL_ID` programatically and store it somewhere to keep track of this initial join message. 

`on_raw_reaction_add` and `on_raw_reaction_remove` should provide the appropriate logic to get and set the `Rubber Duck Opt Out` role for users who do not wish to participate in Rubber Duck Debugging events.

Reply:
I understand your time restrictions, definitely don't stress over this

> Optimally we should fetch the value JOIN_MESSAGE_CHANNEL_ID programatically and store it somewhere to keep track of this initial join message.

I agree. As long as it works as described, the implementation really is up in the air.